### PR TITLE
Moving from Travis to GitHub Actions

### PR DESCRIPTION
**Pull request by davidlutta**
Closes: #264 

## Changes and implications
- Added GitHub action to run tests on push or a pull request 
- Changing the testing badge in the README.md to reflect GitHub actions
- Deleted the travis.yml file

#### Future recommendation
- Cache some directories

### CODE REVIEW

**Review by ErikSchierboom — APPROVED** [review]
LGTM

File: .github/workflows/test.yml
Comment (devkabiir) [inline .github/workflows/test.yml:19]:
```suggestion
      run: pub run bin/check_formatting.dart
```

**Review by devkabiir — APPROVED** [review]
Thank you for your contribution :rocket:

File: .github/workflows/test.yml
Comment (devkabiir) [inline .github/workflows/test.yml:23]:
```suggestion

```

**Comment by devkabiir** [comment]
@ErikSchierboom It looks like something's wrong with either travis or GHA. Do you know anything about this? I can't see the new GHA kicking-in in this PR and merging is blocked by travis.

**Comment by ErikSchierboom** [comment]
Hmmm, looks like it still considers Travis to be required. @iHiD could you merge this?

**Comment by cmccandless** [comment]
Two issues going on here:
- GHA won't run workflows from forks that don't exist yet in the base repository. I suggest opening a new PR to test _after_ this has been merged.
- Travis is still set required; only an admin can disable it as a required check (@iHiD, I know you were pinged above but maybe this got lost in the shuffle?)

**Comment by iHiD** [comment]
Should be mergable now. I also need to turn **on** the GHA check once this is merged. Please remind me.

**Comment by Stargator** [comment]
@davidlutta Thanks for your work. Would you be interested in updating the workflow to include builds for the latest stable version of Dart and the dev version?

That helps us catch any breaking changes that might be coming down the pipe. I can create an issue for it, if you'd like

**Comment by davidlutta** [comment]
@Stargator Yeah sure I'd like to have a go on it.

**Comment by Stargator** [comment]
@davidlutta You can find the issue at #270

**Pull request merged**
